### PR TITLE
Attempts to remove special-casing in brave integration tests

### PR DESCRIPTION
**Pull request by adriancole**
There's one glitch left to analyze cc @trustin @kojilin 

See #2092

### CODE REVIEW

File: brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java [hunk 1]
```diff
@@ -74,7 +74,11 @@ public String path() {
         @Override
         @Nullable
         public String url() {
-            return SpanTags.generateUrl(ctx.log());
+            // This dies because lack of RequestLogAvailability.REQUEST_HEADERS
```
Comment (adriancole) [inline brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java]: This is the glitch. Note: the ServiceRequestContextAdapter works fine as-is

File: brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java [hunk 2]
```diff
@@ -74,7 +74,11 @@ public String path() {
...
+            //return SpanTags.generateUrl(ctx.log());
+
+            // This doesn't die
+            return ((HttpRequest) ctx.request()).uri().toString();
```
Comment (adriancole) [inline brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java]: PS I don't mind doing this instead, just the casting seems accidental/awkward

Comment (kojilin) [reply brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java]: IIRC `ClientRequestContext#request` returns `RpcRequest` when it's RpcClient.

Comment (kojilin) [reply brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java]: IIRC `ctx.request` returns `RpcRequest` when it's a rpc client.

**Comment by adriancole** [comment]
I see. anyway I am flexible on how. would be great to have this data
available without casting etc as it allows secondary sampling use case
client side. server side works already and is more important, but yeah this
is something we should try to formalize.

File: brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java (hunk 2 above)
Comment (adriancole) [reply brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java]: ok so maybe we should add a test which ends up returning an RpcRequest to ensure we don't break it. Do you feel up to adding a commit for that?

Comment (kojilin) [reply brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java]: ok,  I will make one in another PR.

Comment (trustin) [reply brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java]: I'm thinking of splitting `request()` into `request(): HttpRequest` and `rpcRequest(): RpcRequest`, which is similar to @anuraaga's suggestion. Please stay tuned.

**Comment by trustin** [comment]
Closing in favor of #2120

**Comment by trustin** [comment]
Thanks @adriancole for the idea!

File: brave/src/test/java/com/linecorp/armeria/client/brave/BraveClientIntegrationTest.java [hunk 3]
```diff
@@ -94,6 +79,16 @@ public void setupServer() {
         server.setProtocols(protocols);
     }
 
+    @Before
```
Comment (adriancole) [inline brave/src/test/java/com/linecorp/armeria/client/brave/BraveClientIntegrationTest.java]: @trustin I think this is the secret of the incas for your PR

**Pull request closed without merging**